connectors/rdbms/r2dbc/src/main/kotlin/io/github/gabfssilva/river/r2dbc/R2dbcFlowExt.kt

```diff
@@ -0,0 +1,55 @@
+@file:OptIn(FlowPreview::class)
+
+package io.github.gabfssilva.river.r2dbc
+
+import io.github.gabfssilva.river.core.mapParallel
+import io.r2dbc.spi.Connection
+import io.r2dbc.spi.Statement
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flattenConcat
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.reactive.asFlow
+
+
+typealias ResultRow = Map<String, Any?>
+
+fun Connection.query(
+    sql: String
+): Flow<ResultRow> =
+    createStatement(sql)
+        .execute()
+        .asFlow()
+        .map {
```

Comment:
I think you can use `flatMapConcat` directly here instead of `map` + `flattenConcat`.